lib/MetaCPAN/Script/Debian.pm

```diff
@@ -0,0 +1,182 @@
+package MetaCPAN::Script::Debian;
+
+use Moose;
+use namespace::autoclean;
+
+use DBI;
+use List::MoreUtils qw( uniq );
+use CPAN::DistnameInfo;
+use Email::Simple         ();
+use Email::Sender::Simple ();
+
+use MetaCPAN::Types qw( HashRef Str );
+
+with 'MetaCPAN::Role::Script', 'MooseX::Getopt';
+
+has email_to => (
+    is       => 'ro',
+    isa      => Str,
+    required => 1,
+);
+
+has _host_rx => (
+    is      => 'ro',
+    isa     => Str,
+    lazy    => 1,
+    builder => '_build_host_rx',
+);
+
+sub _build_host_rx {
+    my $self = shift;
+
+    my @cpan_hosts = qw<
+        backpan.cpan.org
+        backpan.perl.org
+        cpan.metacpan.org
+        cpan.noris.de
+        cpan.org
+        cpan.perl.org
+        search.cpan.org
+        www.cpan.org
+        www.perl.com
+    >;
+
+    return
+        '^(https?|ftp)://('
+        . join( '|', map {s/\./\\./r} @cpan_hosts ) . ')/';
+}
+
+sub run {
+    my $self = shift;
+
+    # connect to the database
+    my $dbh
+        = DBI->connect(
+        "dbi:Pg:host=public-udd-mirror.xvm.mit.edu;dbname=udd",
+        'public-udd-mirror', 'public-udd-mirror' );
+
+    # special cases
+    my %skip = ( 'libbssolv-perl' => 1 );
+
+    # multiple queries are needed
+    my @sql = (
+
+        # packages with upstream identified as CPAN
+        q{select u.source, u.upstream_url from upstream_metadata um join upstream u on um.source = u.source where um.key='Archive' and um.value='CPAN'},
+
+        # packages which upstream URL pointing to CPAN
+        qq{select source, upstream_url from upstream where upstream_url ~ '${\$self->_host_rx}'},
+    );
+
+    my %dist;
+    my @failures;
+
+    for my $sql (@sql) {
+        my $sth = $dbh->prepare($sql);
+        $sth->execute();
+
+        # map Debian source package to CPAN distro
+        while ( my ( $source, $url ) = $sth->fetchrow ) {
+            next if $skip{$source};
+            $self->dist_for( $source, $url );
+            if ( my $dist = $self->dist_for( $source, $url ) ) {
+                $dist{$dist} = $source;
+            }
+            else {
+                push @failures => [ $source, $url ],
+                    ;
```

Comment:
That semicolon should be on the previous line.  I think the extra comma is messing things up.